extensions/quartz/deployment/src/main/java/io/quarkus/quartz/deployment/QuartzProcessor.java

```diff
@@ -250,7 +224,20 @@ List<ReflectiveClassBuildItem> reflectiveClasses(QuartzBuildTimeConfig config,
             reflectiveClasses.add(ReflectiveClassBuildItem.builder(Connection.class)
                     .reason(getClass().getName()).methods()
                     .fields().build());
-            reflectiveClasses.add(ReflectiveClassBuildItem.builder(driverDialect.getDriver().get())
+            // Since we determine driver delegate at runtime, we need to register all of them
```

Comment:
Now that I think about it, my change is incorrect.
For a custom driver, we still need to register its class for reflection and if there is none, we should register all of the built-in classes.